packages/protocol/src/utils/TrackerRegistry.ts

```diff
@@ -37,40 +31,27 @@ export class TrackerRegistry<T extends TrackerInfo> {
         return this.records[index]
     }
 
-    getAllTrackers(): T[] {
+    getAllTrackers(): TrackerRecord[] {
         return this.records
     }
 }
 
-async function fetchTrackers(contractAddress: string, jsonRpcProvider: string | ConnectionInfo) {
-    const provider = new JsonRpcProvider(jsonRpcProvider)
-    // check that provider is connected and has some valid blockNumber
-    await provider.getBlockNumber()
-
-    const contract = new Contract(contractAddress, trackerRegistryConfig.abi, provider)
-    // check that contract is connected
-    await contract.addressPromise
-
+async function fetchTrackers(config: SmartContractConfig) {
+    const contract = await initContract(config, trackerRegistryConfig.abi)
     if (typeof contract.getNodes !== 'function') {
-        throw Error(`getNodes function is not defined in smart contract (${contractAddress})`)
+        throw Error(`getNodes function is not defined in smart contract (${config.contractAddress})`)
```

Comment:
It seems that the checking of available functions is a typical use case. Maybe `initContract` could implement this checking if we'd add a third parameter which lists all required functions?

Reply:
Good thought, but let's add it if/when there's more than just these two.